lib/useOnyx.ts

```diff
@@ -204,7 +237,7 @@ function useOnyx<TKey extends OnyxKey, TReturnValue = OnyxValue<TKey>>(key: TKey
             previousValueRef.current = newValueRef.current;
 
             // If the new value is `null` we default it to `undefined` to ensure the consumer gets a consistent result from the hook.
-            resultRef.current = [previousValueRef.current as CachedValue<TKey, TReturnValue>, {status: newFetchStatus ?? 'loaded'}];
+            resultRef.current = [previousValueRef.current, {status: newFetchStatus ?? 'loaded'}];
```

Comment:
```suggestion
            resultRef.current = [previousValueRef.current as TReturnValue | undefined, {status: newFetchStatus ?? 'loaded'}];
```

The necessary assertion I mentioned above